indico/queries/submission.py

```diff
@@ -262,3 +316,53 @@ def requests(self) -> Union[Job, str]:
         yield GenerateSubmissionResult(self.submission_id)
         if self.wait:
             yield JobStatus(id=self.previous.id, wait=True, timeout=self.timeout)
+
+
+class SubmitReview(GraphQLRequest):
+    """
+    Submit an "Auto" Review for a submission. Requires that the submission be in PENDING_AUTO_REVIEW status.
+
+    Args:
+        submission_id (int): Id of submission to submit reviewEnabled for
+
+    Options:
+        changes (dict or JSONString): changes to make to raw predictions
+        rejected (boolean): reject the predictions and place the submission
+            in the review queue. Must be True if $changes not provided
+
+    Returns:
+        Job: A Job that can be watched for status of review
+    """
+
+    query = """
+        mutation SubmitReview($submissionId: Int!, $changes: JSONString, $rejected: Boolean) {
+            submitAutoReview(submissionId: $submissionId, changes: $changes, rejected: $rejected) {
+                jobId
+            }
+        }
+
+    """
+
+    def __init__(
+        self,
+        submission: Union[int, Submission],
+        changes: Dict = None,
+        rejected: bool = False,
+    ):
+        submission_id = submission if isinstance(submission, int) else submission.id
+        if not changes and not rejected:
+            raise IndicoInputError("Must provide changes or reject=True")
+        elif changes and isinstance(changes, dict):
+            changes = json.dumps(changes)
+        super().__init__(
+            self.query,
+            variables={
+                "submissionId": submission_id,
+                "changes": changes,
+                "rejected": rejected,
+            },
+        )
+
+    def process_response(self, response) -> Job:
+        response = super().process_response(response)["submitAutoReview"]
+        return Job(id=response["jobId"])
```

Comment:
Is the posting of reviews async? Is that the merge operation server side?

Reply:
it's async - know it's been a while but we talked about this - needs to be async bc we need to validate the review before saving

Reply:
Ah ok makes sense.